src/components/layer-controller/LayerControllerSubscriber.js

```diff
@@ -6,53 +6,133 @@ import {
+      return loader;
+    }
+    case ('ome-tiff'): {
+      // Fetch offsets for ome-tiff if needed.
+      if ('omeTiffOffsetsUrl' in metadata) {
+        const { omeTiffOffsetsUrl } = metadata;
+        const res = await fetch(omeTiffOffsetsUrl, requestInit);
+        if (res.ok) {
+          const offsets = await res.json();
+          const loader = await createOMETiffLoader({
+            url,
+            offsets,
+            headers: requestInit,
+          });
+          return loader;
+        }
+        throw new Error('Offsets not found but provided.');
+      }
+      const loader = createOMETiffLoader({
+        url,
+        headers: requestInit,
+      });
+      return loader;
+    }
+    default: {
+      throw Error(`Image type (${type}) is not supported`);
+    }
+  }
+}
+
+function publishLayer({ loader, imageData, layerId }) {
+  PubSub.publish(LAYER_ADD, {
+    layerId,
+    loader,
+    layerProps: DEFAULT_LAYER_PROPS,
+  });
+  if (loader.getMetadata) {
+    PubSub.publish(METADATA_ADD, {
+      layerId,
+      layerName: imageData.name,
+      layerMetadata: loader.getMetadata(),
+    });
+  }
+}
+
 function LayerControllerSubscriber({ onReady, removeGridComponent }) {
   const [imageOptions, setImageOptions] = useState(null);
-  const [layers, setLayers] = useState([]);
+  const [layersAndLoaders, setLayersAndLoaders] = useState([]);
   const memoizedOnReady = useCallback(onReady, []);
 
   useEffect(() => {
-    function handleRasterAdd(msg, raster) {
-      setImageOptions(raster.images);
+    async function handleRasterAdd(msg, raster) {
+      // render_layers provides the order for rendering initially.
+      const { images, renderLayers } = raster;
+      setImageOptions(images);
+      if (!renderLayers) {
+        const layerId = String(Math.random());
+        // Midpoint of images list as default image to show.
+        const imageData = images[Math.floor(images.length / 2)];
+        const loader = await initLoader(imageData);
+        publishLayer({ loader, imageData, layerId });
+        setLayersAndLoaders([...layersAndLoaders, { layerId, imageData, loader }]);
+      } else {
+        const newLayersAndLoaders = await Promise.all(renderLayers.map(async (imageName) => {
+          const imageData = images.filter(image => image.name === imageName)[0];
+          const layerId = String(Math.random());
+          const loader = await initLoader(imageData);
+          return { layerId, imageData, loader };
+        }));
+        newLayersAndLoaders.forEach(({ imageData, loader, layerId }) => {
+          publishLayer({ loader, imageData, layerId });
+        });
+        setLayersAndLoaders([...layersAndLoaders, ...newLayersAndLoaders]);
+      }
       PubSub.publish(CLEAR_PLEASE_WAIT, 'raster');
     }
     memoizedOnReady();
     const token = PubSub.subscribe(RASTER_ADD, handleRasterAdd);
     return () => PubSub.unsubscribe(token);
-  }, [memoizedOnReady]);
+  }, [memoizedOnReady, layersAndLoaders]);
```

Comment:
See comment about about `useState` setters. Can remove `layersAndLoaders` from this array.